File: Python_shallow/single_eval.py
import TargetFunctions
from Main import optimize
import sys

alg = str(sys.argv[1])
dim = int(sys.argv[2])

tf = TargetFunctions.sphere


constr = [(-10, 10) for _ in range(dim)]
max_iter = 200


optimize(objective_function=tf, problem_dimension=dim, plot_data=False,
         algorithm=alg, time_eval=False, max_iter=max_iter,
         constraints=constr, constraint_handle='projection')

File: R_shallow/single_eval.R
setwd('C:\\Users\\Lenovo\\Desktop\\Studia\\INZYNIERKA\\OptimizationAlgorithms\\R_shallow')

source("TargetFunctions.R")
source("Main.R")

tf <- sphere
max_iter <- 200

args <- commandArgs(trailingOnly=TRUE)

alg <- 'cmaes'
dim <- 6

constr <- list()

res <- optimize(objective_function=tf,
        problem_dimension=dim,
        algorithm=alg,
        max_iter=max_iter,
        constraints=constr,
        repair_method='projection')